### PR TITLE
More tests

### PR DESCRIPTION
**Pull request by jGleitz**
Some more basic tests.
- Fixed the `checkPitch` command by removing unnecessary `input` argument
- Fix for #114: Give all called commands a number
- Improved the error messages for `InvalidCommandLineArgumentTest`
- Fixed some formatting – sorry for the mess at reviewing

### CODE REVIEW

File: src/final2/subtests/InvalidCommandLineArgumentsTest.java [hunk 1]
```diff
@@ -20,76 +20,105 @@ public InvalidCommandLineArgumentsTest() {
...
+		errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "rule=270-90-315-45.5-90");
+		errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "rule=270-90--315-45-90");
+		errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "rule=270-90-315-45-90-");
+		errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "rule=270-90-315-45-90--");
+		errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "rule=270-90-315-45-90---");
+		errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "rule=-270-90-315-45-90");
+		errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "rule=--270-90-315-45-90");
+		errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "rule=-270-90--315-45-90");
+		errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "rule={270,90,315,45,90}");
+		errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "rule=270-0-315-45-90");
```
Comment (jGleitz) [inline src/final2/subtests/InvalidCommandLineArgumentsTest.java:97]: These two lines are the ones I actually added. But due to formatting, `git` does not recognize that.

File: src/final2/subtests/PraktomatPublicTest.java [hunk 2]
```diff
@@ -12,117 +12,113 @@
 /**
  * Simulates the Praktomat's public test.
  * 
- * @author Martin L�per
+ * @author Martin L�per
  * @version 1.0
  */
 public class PraktomatPublicTest extends LangtonSubtest {
-
```
Comment (jGleitz) [inline src/final2/subtests/PraktomatPublicTest.java:19]: I only applied formatting here and fixed a typo. Once again, formatting breaks `git`

File: src/final2/subtests/ValidCommandLineArgumentsTest.java [hunk 3]
```diff
@@ -6,44 +6,53 @@
...
+	}
+
+	/**
+	 * Asserts that the tested class works when passing a well formed speedup argument
+	 */
+	@Test
+	public void speedupArgumentTest() {
+		noOutputTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "speedup=1");
+		noOutputTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "speedup=99");
+		noOutputTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "speedup=2147483647");
```
Comment (jGleitz) [inline src/final2/subtests/ValidCommandLineArgumentsTest.java:37]: This one's new. The rest is not.

File: src/test/InteractiveConsoleTest.java [hunk 4]
```diff
@@ -52,633 +52,652 @@
...
+        for (String string : strings) {
+            if (resultBuilder.length() > 0) {
+                resultBuilder.append(System.lineSeparator());
+            }
+            resultBuilder.append(string);
+        }
+        return resultBuilder.toString();
+    }
+
+    protected static String joinAsNumberdLines(String[] strings) {
```
Comment (jGleitz) [inline src/test/InteractiveConsoleTest.java]: This method's new

File: src/test/InteractiveConsoleTest.java [hunk 5]
```diff
@@ -52,633 +52,652 @@
...
+        errorMessageBuilder.append("the first error occurred before the first command was run:");
+        errorMessageBuilder.append("\n");
+        initOutputRun.check(result[0], errorMessageBuilder.toString());
+
+        for (int j = 1; j < result.length; j++) {
+            Run run = runs[j - 1];
+            errorMessageBuilder = new StringBuilder();
+            errorMessageBuilder.append(consoleMessage(commands, args0));
+            errorMessageBuilder.append("the first error occurred for command\n");
+            errorMessageBuilder.append("[");
```
Comment (jGleitz) [inline src/test/InteractiveConsoleTest.java]: Adding the command's number is new.

**Comment by pcworld** [comment]
I'm for omitting version numbers in class files, seems kind of pointless to me since we don't really make use of version numbers for the tests.

**Comment by jGleitz** [comment]
Yeah, that's true. Eclipse just automatically adds them for me ;)

File: src/final2/subtests/MoveTest.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+				move(0),
+				checkPitch(inputFile),
+				move(1),
+				checkPitch(new String[] {
+						"03",
+						"0a"
+				}),
+				move(1),
+				checkPitch(new String[] {
+						"03",
```
Comment (pcworld) [inline src/final2/subtests/MoveTest.java]: Shouldn't this be `00`? My understanding is that an ant only changes the color of its new cell after it has stepped forward.

File: src/final2/subtests/MoveTest.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+		String[] inputFile = {
+				"0a",
+				"00"
+		};
+		runs = new Run[] {
+				move(0),
+				checkPitch(inputFile),
+				move(1),
+				checkPitch(new String[] {
+						"03",
```
Comment (pcworld) [inline src/final2/subtests/MoveTest.java]: Shouldn't this be `00`? My understanding is that an ant only changes the color of its new cell after it has stepped forward.
(Update: repositioned this comment after the first move(1) command)

Comment (jGleitz) [inline src/final2/subtests/MoveTest.java]: Absolutely right. Always got that wrong when reading the description :/

Comment (pcworld) [inline src/final2/subtests/MoveTest.java]: Yeah, the sheet is very confusing (how it mixes up "Zug" and "Schritt" etc.).
Did the Praktomat's lazyAnt and sportyAnt tests succeed with your interpretation? I tried to replicate your interpretation, but then these two tests fail for me (all others still succeed).

File: src/test/InteractiveConsoleTest.java [hunk 8]
```diff
@@ -111,12 +111,28 @@
...
+			if (resultBuilder.length() > 0) {
+				resultBuilder.append(System.lineSeparator());
+			}
+			resultBuilder.append(string);
 		}
-		return result;
+		return resultBuilder.toString();
+	}
+
+	protected static String joinAsNumberdLines(String[] strings) {
```
Comment (pcworld) [inline src/test/InteractiveConsoleTest.java]: Number**e**d

**Comment by pcworld** [comment]
Otherwise, lgtm.
Btw, you can still review the pull request ignoring the formatting changes: Use [Meld](http://meldmerge.org/) with `git difftool -d`, and enable the "All whitespace" "Text Filter" in Meld's preferences. Edit: Or just use `git diff -b`
(Fixing large chunks of formatting in separate commits would be good nevertheless in the future.)

File: src/final2/subtests/MoveTest.java (hunk 7 above)
Comment (jGleitz) [inline src/final2/subtests/MoveTest.java]: They fail for me anyway because of another bug I'm trying to get a fix on.

**Comment by pcworld** [comment]
can be merged IMO.

**Comment by MartinLoeper** [comment]
yes, they are fine

2015-03-25 12:10 GMT+01:00 pcworld notifications@github.com:

**Pull request merged**